auth/standard_role_manager.hh

```diff
@@ -38,6 +38,8 @@ class standard_role_manager final : public role_manager {
     future<> _stopped;
     abort_source _as;
     std::string _superuser;
+    std::atomic_bool _superuser_created;
```

Comment:
1. why you need atomic_bool, are you accessing it via different shards concurrently? this doesn't seem right. btw there is no atomic_bool usage in the whole codebase, this may be a hint.
2. _superuser_created(this_shard_id() != 0) - it doesn't make sense to fake that if you are not shard then superuser is created, this is misleading. What we usually do is that we explicitly check this_shard_id in the code, and also comment if necessary